gateleen-kafka/src/main/java/org/swisspush/gateleen/kafka/KafkaProducerRecordBuilder.java

```diff
@@ -0,0 +1,101 @@
+package org.swisspush.gateleen.kafka;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import org.swisspush.gateleen.validation.ValidationException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Creates {@link KafkaProducerRecord}s by parsing the request payload.
+ *
+ * @author https://github.com/mcweba [Marc-Andre Weber]
+ */
+class KafkaProducerRecordBuilder {
+
+    private static final String RECORDS = "records";
+    private static final String KEY = "key";
+    private static final String VALUE = "value";
+    private static final String HEADERS = "headers";
+
+    /**
+     * Builds a list of {@link KafkaProducerRecord}s based on the provided payload.
+     * When the payload is not valid (because of missing properties, wrong types, etc.), a {@link ValidationException}
+     * will be thrown holding a description of the error.
+     *
+     * @param topic the kafka topic
+     * @param payload the payload to build the {@link KafkaProducerRecord}s from
+     * @return A list of {@link KafkaProducerRecord}s created from the provided payload
+     * @throws ValidationException when the payload is not valid (missing properties, wrong types, etc.)
+     */
+    static List<KafkaProducerRecord<String, String>> buildRecords(String topic, Buffer payload) throws ValidationException {
+        List<KafkaProducerRecord<String, String>> kafkaProducerRecords = new ArrayList<>();
+        JsonObject payloadObj;
+        try {
+            payloadObj = new JsonObject(payload.toString());
```

Comment:
Is it by intent using `toString()` with implicit charset?

Reply:
The question should be: Why using _String_ at all. We can use the _Buffer_ directly. Will change this.